govstack_api/middleware.py

```diff
@@ -0,0 +1,89 @@
+import os
+from functools import wraps
+from types import SimpleNamespace
+from unittest import mock
+
+from graphene import Schema
+from graphene.test import Client
+
+from django.apps import apps
+from django.contrib.auth import get_user_model
+from django.http import HttpResponse
+from core.schema import Query as CoreQuery, Mutation as CoreMutation
+
+
+def authenticate_decorator(view_func):
+    @wraps(view_func)
+    def _wrapped_view(request, *args, **kwargs):
+        config = apps.get_app_config('govstack_api')
+        im_client = config.IM_CLIENT
+        im_header = request.META.get('HTTP_INFORMATION_MEDIATOR_CLIENT')
+        if im_header != im_client:
+            return HttpResponse('Unauthorized', status=401)
+        else:
+            # self.authenticate_user(request)
+            pass
+
+        return view_func(request, *args, **kwargs)
+    return _wrapped_view
+
+
+class InformationMediatorMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        if request.headers.get('Information-Mediator-Client', None):
+            config = apps.get_app_config('govstack_api')
+            im_client = config.IM_CLIENT
+            im_header = request.META.get('HTTP_INFORMATION_MEDIATOR_CLIENT')
+            if im_header != im_client:
+                return HttpResponse('Unauthorized', status=401)
+            else:
+                self.authenticate_user(request)
+        response = self.get_response(request)
+        return response
+
+    def get_client(self, schema, query, mutation):
+        return Client(schema=schema(query=query, mutation=mutation))
+
+    def create_base_context(self):
+        user = mock.Mock(is_anonymous=False)
+        user.has_perm = mock.MagicMock(return_value=False)
```

Comment:
Same as for gql client, this user is always replaced and it can be removed

Reply:
Yeah I will change it to:
```
    def get_context(self, request):
        if hasattr(request, 'user') and request.user.is_authenticated:
            context = SimpleNamespace(user=request.user)
        else:
            context = SimpleNamespace(user=None)
        return context
```